dms/models/ir_attachment.py

```diff
@@ -0,0 +1,64 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _get_dms_directories(self, res_model, res_id):
+        return self.env["dms.directory"].search(
+            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        )
+
+    def _dms_operations(self, attachment_ids):
+        for attachment in attachment_ids:
```

Comment:
This is not IDs, but recordset. Please name properly variables, or later these names will provoke code errors. Anyway, you are already on the proper model, so no need of extra argument (see the later suggestion):
```suggestion
    def _dms_operations(self):
        for attachment in self:
```